test/ng/directive/ngOptionsSpec.js

```diff
@@ -0,0 +1,1731 @@
+'use strict';
+
+describe('ngOptions', function() {
+
+  var scope, formElement, element, $compile;
+
+  function compile(html) {
+    formElement = jqLite('<form name="form">' + html + '</form>');
+    element = formElement.find('select');
+    $compile(formElement)(scope);
+    scope.$apply();
+  }
+
+  function setSelectValue(selectElement, optionIndex) {
+    var option = selectElement.find('option').eq(optionIndex);
+    selectElement.val(option.val());
+    browserTrigger(element, 'change');
+  }
+
+
+  beforeEach(function() {
+    this.addMatchers({
+      toEqualSelectValue: function(value, multiple) {
+        var errors = [];
+
+        if (multiple) {
+          value = value.map(function(val) { return hashKey(val); });
+        } else {
+          value = hashKey(value);
+        }
+
+        if (!equals(this.actual.val(), value)) {
+          errors.push('Expected select value "' + this.actual.val() + '" to equal "' + hashKey(value) + '"');
```

Comment:
Another good spot! Originally I was hashing inside the if statement but not now.